From fe770c8caf3214554346b71fcfcce810a9c5dd6d Mon Sep 17 00:00:00 2001
From: tinywrkb <tinywrkb@gmail.com>
Date: Mon, 21 Mar 2022 20:12:21 +0200
Subject: [PATCH] Use Meson build system

- Initial support for building with meson
- Remove install script
- Add an AppStream metainfo file. A release tag will need to be added for
  each new version.
- Rename the desktop file and icon to use a reverse DNS naming scheme with
  the same value as the Flathub app.

Closes #18.
---
 GPUViewer                                     |  3 -
 ....github.arunsivaramanneo.GPUViewer.desktop |  2 +-
 ...ub.arunsivaramanneo.GPUViewer.metainfo.xml | 39 +++++++++++
 data/meson.build                              |  7 ++
 gpu-viewer.in                                 |  4 ++
 install                                       | 66 -------------------
 meson.build                                   | 24 +++++++
 7 files changed, 75 insertions(+), 70 deletions(-)
 delete mode 100755 GPUViewer
 rename gpu-viewer.desktop => data/io.github.arunsivaramanneo.GPUViewer.desktop (86%)
 mode change 100755 => 100644
 create mode 100644 data/io.github.arunsivaramanneo.GPUViewer.metainfo.xml
 create mode 100644 data/meson.build
 create mode 100755 gpu-viewer.in
 delete mode 100755 install
 create mode 100644 meson.build

diff --git a/GPUViewer b/GPUViewer
deleted file mode 100755
index d49e544..0000000
--- a/GPUViewer
+++ /dev/null
@@ -1,3 +0,0 @@
-#!/bin/bash
-cd /usr/share/gpu-viewer/Files/
-python3 GPUViewer.py
diff --git a/gpu-viewer.desktop b/data/io.github.arunsivaramanneo.GPUViewer.desktop
old mode 100755
new mode 100644
similarity index 86%
rename from gpu-viewer.desktop
rename to data/io.github.arunsivaramanneo.GPUViewer.desktop
index 82a5ea6..3414fa0
--- a/gpu-viewer.desktop
+++ b/data/io.github.arunsivaramanneo.GPUViewer.desktop
@@ -4,7 +4,7 @@ Version=1.5
 Name=GPU-Viewer
 GenericName=Utility
 Comment=A front-end to glxinfo, vulkaninfo, clinfo and es2_info
-Icon=/usr/share/gpu-viewer/Images/GPU_Viewer.png
+Icon=io.github.arunsivaramanneo.GPUViewer
 Exec=gpu-viewer
 Terminal=false
 Categories=System;GTK;GNOME
diff --git a/data/io.github.arunsivaramanneo.GPUViewer.metainfo.xml b/data/io.github.arunsivaramanneo.GPUViewer.metainfo.xml
new file mode 100644
index 0000000..d5e4a44
--- /dev/null
+++ b/data/io.github.arunsivaramanneo.GPUViewer.metainfo.xml
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+    <id>io.github.arunsivaramanneo.GPUViewer</id>
+    <launchable type="desktop-id">io.github.arunsivaramanneo.GPUViewer.desktop</launchable>
+    <name>GPU-Viewer</name>
+    <summary>A front-end to glxinfo, vulkaninfo, clinfo and es2_info</summary>
+    <description>
+        <p>
+            This project aims to capture all the important details of glxinfo, vulkaninfo and clinfo in a GUI. The project is being developed using python 3 pygobject with GTK3. All the important details are extracted using glxinfo/vulkaninfo/clinfo with the combination of grep, CAT , AWK commands and displayed in the front-end. There is no hard OpenGL Programming involved, until glxinfo, vulkaninfo and clinfo works the GPU-viewer will also work.
+        </p>
+    </description>
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>GPL-3.0</project_license>
+    <url type="homepage">https://github.com/arunsivaramanneo/GPU-Viewer</url>
+    <url type="bugtracker">https://github.com/arunsivaramanneo/GPU-Viewer/issues</url>
+    <url type="donation">https://www.paypal.me/ArunSivaraman</url>
+    <screenshots>
+        <screenshot type="default">
+            <image>https://user-images.githubusercontent.com/30646692/42731013-13f37576-8822-11e8-96c8-04ab062f690f.png</image>
+        </screenshot>
+        <screenshot>
+            <image>https://user-images.githubusercontent.com/30646692/42731014-157f0b26-8822-11e8-95a8-695852d1d7c1.png</image>
+        </screenshot>
+        <screenshot>
+            <image>https://user-images.githubusercontent.com/30646692/42731015-174af686-8822-11e8-8dfb-5baad1cface1.png</image>
+        </screenshot>
+        <screenshot>
+            <image>https://user-images.githubusercontent.com/30646692/42731016-18ca0628-8822-11e8-90c1-def21ee5d0bb.png</image>
+        </screenshot>
+    </screenshots>
+    <content_rating type="oars-1.1"/>
+    <releases>
+        <release version="1.38" date="2022-03-21"/>
+        <release version="1.37" date="2021-12-06"/>
+        <release version="1.36" date="2021-10-15"/>
+        <release version="1.35" date="2021-09-27"/>
+        <release version="1.15" date="2018-12-22"/>
+    </releases>
+</component>
diff --git a/data/meson.build b/data/meson.build
new file mode 100644
index 0000000..ea57dff
--- /dev/null
+++ b/data/meson.build
@@ -0,0 +1,7 @@
+install_data(APPLICATION_ID + '.desktop',
+  install_dir: get_option('datadir') + '/applications')
+install_data(APPLICATION_ID + '.metainfo.xml',
+  install_dir: get_option('datadir') + '/metainfo')
+install_data('../Images/GPU_Viewer.png',
+  rename: APPLICATION_ID + '.png',
+  install_dir: get_option('datadir') + '/icons/hicolor/512x512/apps')
diff --git a/gpu-viewer.in b/gpu-viewer.in
new file mode 100755
index 0000000..2428d1b
--- /dev/null
+++ b/gpu-viewer.in
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+cd @prefix@/@datadir@/gpu-viewer/Files
+exec python3 GPUViewer.py
diff --git a/install b/install
deleted file mode 100755
index d53eb18..0000000
--- a/install
+++ /dev/null
@@ -1,66 +0,0 @@
-#!/bin/bash
-echo installation starts...
-echo " "
-lsb_release -a
-user=$(whoami)
-echo ""
-file1="/usr/share/gpu-viewer"
-file4="/usr/bin/vulkaninfo"
-file5="/home/$user/Desktop/gpu-viewer.desktop"
-file6="/usr/bin/gpu-viewer"
-echo "************************GPU Viewer 1.38 - copyright(c) 2017-2021 - Arun Sivaraman****************************"
-if [ ! -d "$file1" ]
-then
-
-	sudo mkdir /usr/share/gpu-viewer
-fi
-echo " "
-echo "Installing Application files ...."
-echo " "
-sudo cp -vrf Files /usr/share/gpu-viewer
-echo " "
-echo "Application Files Successfully Installed...."
-echo " "
-echo "Installing Application Icons ...."
-echo " "
-sudo cp -vrf Images /usr/share/gpu-viewer
-echo " "
-echo "Application Icons Succesfully Installed ...."
-echo ""
-echo "Installing Supported Documents...."
-echo " "
-sudo cp -vrf About\ GPU\ Viewer /usr/share/gpu-viewer
-sudo cp -vrf GPUViewer /usr/share/gpu-viewer
-sudo cp -vrf Change\ Log /usr/share/gpu-viewer
-sudo cp -vrf LICENSE /usr/share/gpu-viewer
-sudo cp -vrf README.md /usr/share/gpu-viewer
-echo " "
-echo "Supporting documents Successfully Installed...."
-echo " "
-echo "Installing files completed..."
-echo " "
-sudo cp gpu-viewer.desktop /usr/share/applications/
-echo " "
-echo "Application successfully added to the menu"
-if [ ! -f "$file6" ]
-then
-    sudo ln -s /usr/share/gpu-viewer/GPUViewer /usr/bin/gpu-viewer
-    echo "Symlink created - Application can be accessed from Terminal, enter gpu-viewer ENTER "
-fi
-echo " "
-if [ ! -f "$file5" ]
-then
-	read -p "Do you want to add a shortcut to desktop(y/n):" choice
-    echo " "
-	if [ $choice == 'y' ]
-	then
-    	sudo cp gpu-viewer.desktop /home/$user/Desktop/
-	    sudo chown -R $user /home/$user/Desktop/gpu-viewer.desktop
-    	echo "Desktop Shortcut successfully created ..."
-        echo " "
-	fi
-fi
-echo Installation complete...
-echo "All done !!! Enjoy the Application"
-echo "Consider supporting this application by giving a star on my Github repository (https://github.com/arunsivaramanneo/GPU-Viewer) or by donating (https://www.paypal.me/ArunSivaraman)"
-date
diff --git a/meson.build b/meson.build
new file mode 100644
index 0000000..0c1edd1
--- /dev/null
+++ b/meson.build
@@ -0,0 +1,24 @@
+project('gpu-viewer',
+  license: 'GPL-3.0-or-later'
+)
+
+APPLICATION_ID = 'io.github.arunsivaramanneo.GPUViewer'
+
+subdir('data')
+bin_conf = configuration_data()
+bin_conf.set('prefix', get_option('prefix'))
+bin_conf.set('datadir', get_option('datadir'))
+
+configure_file(
+  input: 'gpu-viewer.in',
+  output: 'gpu-viewer',
+  configuration: bin_conf,
+  install_dir: get_option('bindir')
+)
+
+install_data([ 'About GPU Viewer', 'Change Log.md', 'LICENSE', 'README.md' ])
+
+install_subdir('Files',
+  install_dir: get_option('datadir') + '/gpu-viewer')
+install_subdir('Images',
+  install_dir: get_option('datadir') + '/gpu-viewer')
-- 
2.35.1

